homeassistant/components/isy994/button.py

```diff
@@ -80,3 +100,38 @@ def __init__(self, node: Node, base_unique_id: str) -> None:
     async def async_press(self) -> None:
         """Press the button."""
         await self._node.beep()
+
+
+class ISYNetworkResourceButtonEntity(ButtonEntity):
+    """Representation of an ISY/IoX Network Resource button entity."""
+
+    _attr_should_poll = False
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_has_entity_name = True
+
+    def __init__(self, node: Node, base_unique_id: str) -> None:
+        """Initialize a beep Insteon device button entity."""
+        self._node = node
+
+        # Entity class attributes
+        self._attr_name = node.name
+        self._attr_unique_id = f"{base_unique_id}_{node.address}"
+        url = _async_isy_to_configuration_url(node.isy)
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (
+                    ISY994_DOMAIN,
+                    f"{node.isy.configuration[ISY_CONF_UUID]}_{PROTO_NETWORK_RESOURCE}",
+                )
+            },
+            manufacturer=MANUFACTURER,
+            name=f"{node.isy.configuration[ISY_CONF_NAME]} {ISY_CONF_NETWORKING}",
+            model=node.isy.configuration[ISY_CONF_MODEL],
+            sw_version=node.isy.configuration[ISY_CONF_FIRMWARE],
+            configuration_url=url,
+            via_device=(ISY994_DOMAIN, node.isy.configuration[ISY_CONF_UUID]),
```

Comment:
You might be able to save some line length with `configuration = node.isy.configuration` above

Reply:
Updated